types/header.go

```diff
@@ -89,8 +92,15 @@ func (h *Header) Verify(untrst header.Header) error {
 	// perform actual verification
 	if untrstH.Height() == h.Height()+1 {
 		// Check the validator hashes are the same in the case headers are adjacent
-		// TODO: next validator set is not available
 		if !bytes.Equal(untrstH.AggregatorsHash[:], h.AggregatorsHash[:]) {
+			return &header.VerifyError{
+				Reason: fmt.Errorf("expected old header validators (%X) to match those from new header (%X)",
+					h.AggregatorsHash,
+					untrstH.AggregatorsHash,
+				),
+			}
+		}
+		if !bytes.Equal(untrstH.NextAggregatorsHash[:], h.NextAggregatorsHash[:]) {
```

Comment:
should this check be between `untrstH. AggregatorsHash [:]` and `h.NextAggregatorsHash[:]`? you want to compare and make sure that the previous header next validators hash is equal to current validator hash?

Reply:
Good point!

Reply:
Comparing `untrstH.AggregatorsHash` against `h.NextAggregatorsHash` now.